src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -2892,7 +2892,13 @@ public void Dispose()
             dynamoViewModel.SideBarTabItems.CollectionChanged -= this.OnCollectionChanged;
 
             if (fileTrustWarningPopup != null)
+            {
                 fileTrustWarningPopup.CleanPopup();
+            }
+            //TODO code smell.
+            var workspaceView = this.ChildOfType<WorkspaceView>();
+            workspaceView?.Dispose();
+            (workspaceView?.NodeAutoCompleteSearchBar?.Child as IDisposable)?.Dispose();
```

Comment:
Why do we need to dispose this specific control?

Reply:
the `NodeAutoCompleteSearchControl` has a dispose method which does not seem to be called from anywhere else.

Reply:
Minor point but how about disposing it inside the workspaceview's dispose method - would be cleaner.